lib/util/git/index.ts

```diff
@@ -595,6 +598,7 @@ export async function isBranchBehindBase(
       { currentBranch, currentBranchSha },
       `branch.isBehindBase(): ${isBehind}`
     );
+    setCachedBehindBaseResult(branchName, isBehind);
```

Comment:
This is a bug fix, it could be separated if desired but is highly relevant to this PR because it wasn't causing a _functional_ problem previously (maybe performance, we'll see)

Reply:
Give me a min. I will check why this wasn't added by me previously.

Reply:
https://github.com/renovatebot/renovate/pull/17848
Seems it got overlooked cause there were too many changes in this PR.